analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StreamReadStatement.cs

```diff
@@ -112,15 +112,17 @@ public async Task Foo()
         {
             var stream = new DerivedStream();
             var array = new byte[10];
-            stream.Read(array, 0, "");                                                    // Compliant
-            stream.Read(array, 0, 10);                                                    // Noncompliant
-            stream.ReadAsync(array, 0, (int)stream.Length);                               // Noncompliant {{Check the return value of the 'ReadAsync' call to see how many bytes were read.}}
-            await stream.ReadAsync(array, 0, (int)stream.Length);                         // Noncompliant
-            await stream.ReadAsync(array, 0, (int)stream.Length).ConfigureAwait(false);   // Noncompliant
-            await (stream.ReadAsync(array, 0, (int)stream.Length)).ConfigureAwait(false); // Noncompliant
-            await (stream.ReadAsync(array, 0, (int)stream.Length));                       // Noncompliant
-            var res = await stream.ReadAsync(array, 0, (int)stream.Length);               // Compliant
-            var x = stream.ReadAsync(array, 0, (int)stream.Length).Result;                // Compliant
+            stream.Read(array, 0, "");                                                     // Compliant
+            stream.Read(array, 0, 10);                                                     // Noncompliant
+            stream.ReadAsync(array, 0, (int)stream.Length);                                // Noncompliant {{Check the return value of the 'ReadAsync' call to see how many bytes were read.}}
+            await stream.ReadAsync(array, 0, (int)stream.Length);                          // Noncompliant
+            await stream.ReadAsync(array, 0, (int)stream.Length).ConfigureAwait(false);    // Noncompliant
+            await (stream.ReadAsync(array, 0, (int)stream.Length)).ConfigureAwait(false);  // Noncompliant
+            await (stream.ReadAsync(array, 0, (int)stream.Length));                        // Noncompliant
+            var res = await stream.ReadAsync(array, 0, (int)stream.Length);                // Compliant
+            var t1 = stream.ReadAsync(array, 0, (int)stream.Length);                       // Compliant The read length is stored in the task and that is as good as storing the read length in a variable
+            var t2 = stream.ReadAsync(array, 0, (int)stream.Length).ConfigureAwait(false); // Compliant
```

Comment:
`t1` and `t2` are the actually added test cases (Use Githubs "Hide Whitespace" feature to confirm).